rosbag2_transport/src/rosbag2_transport/generic_publisher.hpp

```diff
@@ -28,8 +28,9 @@ class GenericPublisher : public rclcpp::PublisherBase
 public:
   GenericPublisher(
     rclcpp::node_interfaces::NodeBaseInterface * node_base,
-    const std::string & topic,
-    const rosidl_message_type_support_t & type_support);
+    const rosidl_message_type_support_t & type_support,
+    const std::string & topic_name,
+    const rclcpp::QoS & qos);
```

Comment:
Is this change to the API acceptable?

Reply:
This is an internal API (notice `src/` directory) - I changed the argument order to be consistent with (GenericSubscription)[https://github.com/ros2/rosbag2/pull/337/files?file-filters%5B%5D=.cpp&file-filters%5B%5D=.hpp#diff-edcbd943e37cb6ab8b38a9e64d0d46feR48] because it was just confusing that they were not in the same order